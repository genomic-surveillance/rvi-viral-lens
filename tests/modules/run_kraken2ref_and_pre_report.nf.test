nextflow_process {

    name "Test Process run_kraken2ref_and_pre_report process"
    script "modules/run_kraken2ref_and_pre_report.nf"
    process "run_k2r_sort_reads"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                meta = [
                    id:"test.9000000",
                    sample_id:"test",
                    taxid:9000000
                ]
                test_loc="${projectDir}/tests/test_data/"
            }

            process {

            """
            // define inputs of the process here. Example:
            input[0] = tuple(
                params.meta, 
                file("${params.test_loc}/test_kraken_outputs/test.kraken.output"),
                file("${params.test_loc}/test_kraken_outputs/test.report.txt"))
            """

            }
        }

        then {
            assert process.success
            
            // NOTE: the "_decomposed.json" now have timestamps, which invalidates
            // checking snapshots. For the future, we should consider to check md5
            // under groovy and remove the time stamp before that.
            
            //assert snapshot(process.out).match()
        }
    }
}
