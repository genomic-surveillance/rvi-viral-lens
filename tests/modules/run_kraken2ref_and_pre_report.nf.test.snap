{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test.9000000",
                            "taxid": 9000000,
                            "sample_id": "test"
                        },
                        "test.9000000_tax_to_reads.json:md5,2e80bd3fe33676daed6854a9093ebdb8",
                        "test.9000000_decomposed.json:md5,cc6f87a42369e2f801bed0704e198b3d"
                    ]
                ],
                "json_files": [
                    [
                        {
                            "id": "test.9000000",
                            "taxid": 9000000,
                            "sample_id": "test"
                        },
                        "test.9000000_tax_to_reads.json:md5,2e80bd3fe33676daed6854a9093ebdb8",
                        "test.9000000_decomposed.json:md5,cc6f87a42369e2f801bed0704e198b3d"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-20T14:01:43.675223"
    }
}